tests/sync/sync_manager.cpp

```diff
@@ -117,19 +118,19 @@ TEST_CASE("sync_manager: user state management", "[sync]") {
     const std::string identity_3 = "user-baz";
 
     SECTION("should get all users that are created during run time") {
-        SyncManager::shared().get_user(identity_1, r_token_1, a_token_1, url_1);
-        SyncManager::shared().get_user(identity_2, r_token_2, a_token_2, url_2);
+        SyncManager::shared().get_user(identity_1, r_token_1, a_token_1, url_1, dummy_device_id);
+        SyncManager::shared().get_user(identity_2, r_token_2, a_token_2, url_2, dummy_device_id);
         auto users = SyncManager::shared().all_users();
         REQUIRE(users.size() == 2);
         CHECK(validate_user_in_vector(users, identity_1, url_1, r_token_1, a_token_1));
         CHECK(validate_user_in_vector(users, identity_2, url_2, r_token_2, a_token_2));
     }
 
     SECTION("should be able to distinguish users based solely on URL") {
-        SyncManager::shared().get_user(identity_1, r_token_1, a_token_1, url_1);
-        SyncManager::shared().get_user(identity_1, r_token_1, a_token_1, url_2);
-        SyncManager::shared().get_user(identity_1, r_token_1, a_token_1, url_3);
-        SyncManager::shared().get_user(identity_1, r_token_1, a_token_1, url_1); // existing
+        SyncManager::shared().get_user(identity_1, r_token_1, a_token_1, url_1, dummy_device_id);
+        SyncManager::shared().get_user(identity_1, r_token_1, a_token_1, url_2, dummy_device_id);
+        SyncManager::shared().get_user(identity_1, r_token_1, a_token_1, url_3, dummy_device_id);
+        SyncManager::shared().get_user(identity_1, r_token_1, a_token_1, url_1, dummy_device_id); // existing
         auto users = SyncManager::shared().all_users();
         REQUIRE(users.size() == 3);
         CHECK(validate_user_in_vector(users, identity_1, url_1, r_token_1, a_token_1));
```

Comment:
`validate_user_in_vector` should be changed to accept device_id, which should then be validated. This should be changed everywhere in this file.

Reply:
Updated this method in tests